frontend/app/src/components/bots/BotBuilder.svelte

```diff
@@ -33,12 +38,37 @@
     let errors = $derived.by(
         debouncedDerived(
             () => [$state.snapshot(candidate)],
-            () => validateBot(candidate),
+            async () => {
+                const errors = validateBot(candidate);
+                if (errors.get("bot_name").length == 0) {
+                    errors.addErrors("bot_name", await checkUsername(candidate.name));
+                }
+                return errors;
+            },
             300,
             new ValidationErrors(),
         ),
     );
 
+    function checkUsername(value: string): Promise<ValidationErrorMessages> {
+        return client
+            .checkUsername(value, true)
```

Comment:
We need a bot specific version of this because bot names can clash with usernames.
We had to allow this because the register bot proposals can take a few days and anyone in that time could register the same name and cause them to fail.

Reply:
@hpeebles checkUsernme takes a bool now which means it can be used for users or bots